bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/ConfigUtils.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+    public static boolean containsSupportedMembers(BUnionType unionType) {
+        for (Type memberType : unionType.getMemberTypes()) {
+            if (!isCliSupported(TypeUtils.getImpliedType(memberType).getTag())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static boolean isCliSupported(int tag) {
+        return tag <= TypeTags.BOOLEAN_TAG || TypeTags.isXMLTypeTag(tag);
+    }
+
+    public static Object getUnionValue(VariableKey key, BUnionType unionType, String value, String arg) {
+        List<Object> matchingValues = getConvertibleMemberValues(value, unionType);
+        if (matchingValues.size() == 1) {
+            return matchingValues.get(0);
+        }
+        String typeName = decodeIdentifier(unionType.toString());
+        if (matchingValues.isEmpty()) {
+            throw new ConfigException(CONFIG_INCOMPATIBLE_TYPE, arg, key.variable, typeName, value);
+        }
+        throw new ConfigException(CONFIG_UNION_VALUE_AMBIGUOUS_TARGET, arg, key.variable, typeName);
+    }
+
+    private static List<Object> getConvertibleMemberValues(String value, UnionType unionType) {
+        List<Object> matchingValues = new ArrayList<>();
+        for (Type type : unionType.getMemberTypes()) {
+            switch (TypeUtils.getImpliedType(type).getTag()) {
+                case TypeTags.BYTE_TAG:
+                    convertAndGetValuesFromString(matchingValues, TypeConverter::stringToByte, value);
+                    break;
+                case TypeTags.INT_TAG:
+                    convertAndGetValuesFromString(matchingValues, TypeConverter::stringToInt, value);
+                    break;
+                case TypeTags.BOOLEAN_TAG:
+                    convertAndGetValuesFromString(matchingValues, TypeConverter::stringToBoolean, value);
+                    break;
+                case TypeTags.FLOAT_TAG:
+                    convertAndGetValuesFromString(matchingValues, TypeConverter::stringToFloat, value);
+                    break;
+                case TypeTags.DECIMAL_TAG:
+                    convertAndGetValuesFromString(matchingValues, TypeConverter::stringToDecimal, value);
+                    break;
+                case TypeTags.STRING_TAG:
+                    convertAndGetValuesFromString(matchingValues, StringUtils::fromString, value);
+                    break;
+                default:
+                    convertAndGetValuesFromString(matchingValues, TypeConverter::stringToXml, value);
+            }
+        }
+        return matchingValues;
+    }
+
+    private static void convertAndGetValuesFromString(List<Object> matchingValues,
+                                                      Function<String, Object> convertFunc, String value) {
+        Object unionValue;
+        try {
+            unionValue = convertFunc.apply(value);
+        } catch (NumberFormatException | BError e) {
+            return;
```

Comment:
Will there be any issue with not handling this error?

Reply:
For union type, we keep a `List<Object> matchingValues` to get the convertible values. If we get a `NumberFormatException` or `BError`, we ignore the error and do not add that to the `matchingValues` list. And at the end of the loop, we throw the incompatible type error if the list is empty.
So I think we can ignore the error here because the error denotes the value is not convertible.